packages/datepickers/src/elements/DatepickerRange/components/Month.tsx

```diff
@@ -231,6 +232,37 @@ export const Month = forwardRef<HTMLDivElement, IMonthProps>(
             onClick={() => {
               if (!isDisabled) {
                 dispatch({ type: 'CLICK_DATE', value: date });
+                if (onChange) {
+                  if (state.isStartFocused) {
```

Comment:
The logical branching is pulled from take from `CLICK_DATE` switch case in the reducer function.